pyomo/contrib/doe/doe.py

```diff
@@ -412,66 +361,107 @@ def _sequential_finite(self, read_output, extract_single_model, store_output):
+            scena_gen = scena_object.generate_scenario()
+            
+            # a list of dictionary, each one is a parameter dictionary with perturbed parameter values
+            self.scenario_list = scena_gen["scenario"]
+            # dictionary, keys are parameter name, values are a list of scenario index where this parameter is perturbed.
+            self.scenario_num = scena_gen["scena_num"]
+
             # dict for storing model outputs
             output_record = {}
 
-            # dict for storing Jacobian
-            models = []
-            time_allbuild = []
-            time_allsolve = []
-            # loop over each scenario
-            for no_s in (scena_gen.scena_keys):
-
-                scenario_iter = scena_gen.next_sequential_scenario(no_s)
-                # create the model
-                time0_build = time.time()
-                mod = self.create_model(scenario_iter, args=self.args)
-                time1_build = time.time()
-                time_allbuild.append(time1_build-time0_build)
-
-                # discretize if needed
-                if self.discretize_model:
-                    mod = self.discretize_model(mod)
-
-                # solve model
-                time0_solve = time.time()
-                square_result = self._solve_doe(mod, fix=True)
-                time1_solve = time.time()
-                time_allsolve.append(time1_solve-time0_solve)
-                models.append(mod)
-
-                if extract_single_model:
-                    mod_name = store_output + str(no_s) + '.csv'
-                    dataframe = extract_single_model(mod, square_result)
-                    dataframe.to_csv(mod_name)
+            # Create a global model 
+            mod = pyo.ConcreteModel()
+
+            # Set for block/scenarios
+            mod.scena = pyo.Set(initialize=list(range(len(self.scenario_list))))
+
+            # Allow user to self-define complex design variables
+            self.create_model(mod=mod, model_option="global")
+
+            def block_build(b,s):
+                # create block scenarios
+                self.create_model(mod=b, model_option="block")
+                
+                # fix parameter values to perturbed values
+                for par in self.param:
+                    par_strname = eval('b.'+str(par))
+                    par_strname.fix(scena_gen["scenario"][s][par])
+
+            mod.block = pyo.Block(mod.scena, rule=block_build)
 
+            # discretize the model        
+            if self.discretize_model:
+                mod = self.discretize_model(mod)
+
+            # force all design variables in blocks be the same as global design variables
+            #def fix_design1(m,s):
+            #    return m.block[s].CA0[0]  == m.CA0[0]
+            
+            #def fix_design2(m,s,t):
+            #    return m.block[s].T[t] == m.T[t]
+
+            for name in self.design_name:
```

Comment:
Does this work if the design variable is indexed in time? Is self.design_name a list of something else?

Reply:
yes.